rfcs/text/0352-graph-type-system.md

```diff
@@ -0,0 +1,2585 @@
+>
+> As such, examples are not provided in this section, refer to the Reference-Level description for more information.
+
+#### Describing multiple possibilities
+
+A Property Type can also express that its value is _either_ something _or_ something else, where "something" and "something else" are defined as outlined above using Data Types or an object definition.
+
+**Example 1**
+
+The `User ID` Property Type could define its value as being _either_ an instance of the `Text` Data Type _or_ an instance of the `Number` Data Type
+
+- Sample data when used in (simplified view of) an Entity
+
+  ```json
+  {
+    "userId": 42088130893
+  }
+  ```
+
+  or
+
+  ```json
+  {
+    "userId": "c09b1839-8084-4a2d-9713-5d074c9c6ce2"
+  }
+  ```
+
+**Example 2**
+
+The `Contrived Property` Property Type could define its value as being _either_ an instance of the `Number` Data Type, _or_ an object which has a list with a maximum of 4 `Foo` property values
+
+- Sample data when used in (simplified view of) an Entity
+
+  Assuming that `Foo` accepts an instance of the `Text` Data Type
+
+  ```json
+  {
+    "contrivedProperty": 32
+  }
+  ```
+
+  or
+
+  ```json
+  {
+    "contrivedProperty": {
+      "foo": ["one", "two"]
+    },
+    ...
+  }
+  ```
+
+### Link Types
+
+A **Link Type** is a description of a _directional_ relationship between two things. A Link Type is composed of the following:
+
+- a **required** `$id`, which is a globally unique identifier, where the Property Type's definition can be accessed (in most cases this will be a URL)
+- a **required** `name`, which should be a non-pluralized description of the relationship (for example: "Friend" not "Friends")
+- a **required** `description` to further explain the semantic meaning of the relationship
+- a **possible** `relatedKeywords`, which is a list of terms that are related to the link type
+
+> 💭 As the next section (Entity Types) will outline, there's an additional step to quantifying the types of things on either end of the relationship, as such the descriptions **should** be very general and most of the time not contain specific words about types (such as person).
+
+**Example 1**
+
+There could be an `Owns` link type with a `description` of "Have (something) as one's own; possess"
+
+**Example 2**
+
+There could be a `Submitted By` link type with a `description` of "Suggested, proposed, or presented by"
+
+### Entity Types
+
+An Entity Type is a description of a particular "thing", made up of identifiable pieces of data. An Entity Type is composed of the following:
+
+- a **required** `$id`, which is a globally unique identifier, where the Entity Type's definition can be accessed (in most cases this will be a URL)
+- a **required** `name`, which should generally be a non-pluralized description of the thing
+- **optionally** a `description` to further explain the semantic meaning of the Entity Type
+- a **required** definition of its possible values.
```

Comment:
Feel free to commit, I don't have a strong opinion on this specific phrasing. If you think it improves readability please add it, the following is just more of a thought stream in general on this.

I think generally speaking it might be helpful for us to use "values" more generally than you seem to associate it with. E.g. the "value" of a Property _Type_, etc. (similar to how you used to use "point to" when coming more from a property graph view of things)

And I do think the values of an Entity are defined by the Property Types in the Entity Type, it's just that those Property Types then in turn define their values by Data Types (and other stuff). It all gets collapsed.

Reply:
I personally don't find the idea of a "value" of an entity type very clear, so think being more specific would be helpful here so will commit it. Better understand where you're coming from now though, thanks.